rpt2 does not resolve pnpm symlinked dependencies

#### What happens and why it is wrong

When trying to build a project installed with [`pnpm`](https://pnpm.js.org/), `rpt2` fails to resolve the created symlinked dependencies in `node_modules` (non-flat).

Setting `check: false` is (arguably) a workaround.

Related issues: #188, #214

#### Reproduction

Repository: https://github.com/davelsan/rpt2-pnpm-symlinks
Config: https://github.com/davelsan/rpt2-pnpm-symlinks/blob/master/rollup.config.js

The provided repro is a minimal `vue3` plugin that imports the `App` interface from `vue` in [`index.ts`](https://github.com/davelsan/rpt2-pnpm-symlinks/blob/7345a15cb9b756a6216fa5b3a0772a7a87314a27/src/index.ts#L1)

```ts
import  { App } from 'vue'
```

This import causes the following error on build:

```
 [!] (plugin rpt2) Error: semantic error TS2305: Module '"../node_modules/vue/dist/vue"' has no exported member 'App'.
```

Setting the `check` option to `false` in [`rollup.config.js`](https://github.com/davelsan/rpt2-pnpm-symlinks/blob/7345a15cb9b756a6216fa5b3a0772a7a87314a27/rollup.config.js#L13) eliminates the error.

#### Versions
- typescript: 4.2.3
- rollup: 2.41.5
- rollup-plugin-typescript2: 0.30.0

#### rollup.config.js

```js
import path    from 'path';
import rpt2    from 'rollup-plugin-typescript2';
import pkgJson from './package.json';

export default {

  input: './src/index.ts',

  external: ['vue'],

  plugins: [
    rpt2({
      check: true,
      tsconfig: path.resolve(__dirname, 'tsconfig.json'),
    })
  ],

  output: [
    {
      file: pkgJson.browser, // dist/index.esm.js
      format: 'es',
      sourcemap: true,
    },
  ],
}
```

#### tsconfig.json

A snippet with the relevant module resolution options only.

```json
{
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
  }
}
```

#### package.json

A snippet with the relevant dependencies only.

```json
{
  "peerDependencies": {
    "vue": "^3.0.7"
  },
  "devDependencies": {
    "rollup": "^2.41.5",
    "rollup-plugin-typescript2": "^0.30.0",
    "typescript": "^4.2.3",
    "vue": "^3.0.7"
  }
}
```

Any updates on this? I'm trying to bundle a react app with rollup and the original @rollup/plugin-typescript gives debug failure errors, so I tried this.

I got the following error specifically:

```
[!] (plugin rpt2) Error: /media/theo/TheoData/Essentials/Projects/Programming/hypervpn/src/pages/home.tsx(1,10): semantic error TS2305: Module '"@chakra-ui/react"' has no exported member 'Button'.
src/pages/home.tsx
Error: /media/theo/TheoData/Essentials/Projects/Programming/hypervpn/src/pages/home.tsx(1,10): semantic error TS2305: Module '"@chakra-ui/react"' has no exported member 'Button'.
    at error (/media/theo/TheoData/Essentials/Projects/Programming/hypervpn/node_modules/.pnpm/rollup@2.40.0/node_modules/rollup/dist/shared/rollup.js:5279:30)
```

--- a/__tests__/host.spec.ts
+++ b/__tests__/host.spec.ts
@@ -2,7 +2,7 @@ import { afterAll, beforeAll, test, expect, jest } from "@jest/globals";
 import * as ts from "typescript";
 import * as path from "path";
 import { normalizePath as normalize } from "@rollup/pluginutils";
-import { remove, ensureDir, writeFile } from "fs-extra";
+import { remove, ensureDir, writeFile, ensureSymlink } from "fs-extra";
 
 import { setTypescriptModule } from "../src/tsproxy";
 import { LanguageServiceHost } from "../src/host";
@@ -18,12 +18,14 @@ const unaryFuncSnap = { text: unaryFunc };
 const local = (x: string) => normalize(path.resolve(__dirname, x));
 const testDir = local("__temp/host");
 const testFile = `${testDir}/file.ts`;
+const linkedTestFile = `${testDir}/link.ts`;
 const nonExistent = `${testDir}/this-does-not-exist.ts`;
 
 afterAll(() => remove(testDir));
 beforeAll(async () => {
 	await ensureDir(testDir);
 	await writeFile(testFile, unaryFunc, "utf8");
+	await ensureSymlink(testFile, linkedTestFile);
 });
 
 test("LanguageServiceHost", async () => {
@@ -55,10 +57,12 @@ test("LanguageServiceHost", async () => {
 	expect(host.directoryExists(nonExistent)).toBeFalsy();
 	expect(host.fileExists(nonExistent)).toBeFalsy();
 	expect(host.fileExists(testFile)).toBeTruthy();
-	expect(host.readDirectory(testDir)).toEqual([testFile]);
+	expect(host.readDirectory(testDir)).toEqual([testFile, linkedTestFile]);
 	expect(host.readFile(nonExistent)).toBeFalsy();
 	expect(host.readFile(testFile)).toEqual(unaryFunc);
 	expect(host.useCaseSensitiveFileNames()).toBe(process.platform === "linux");
+	expect(host.realpath(testFile)).toEqual(testFile);
+	expect(host.realpath(linkedTestFile)).toEqual(testFile);
 
 	// test misc functionality
 	expect(host.getCompilationSettings()).toEqual(testOpts);

--- a/__tests__/host.spec.ts
+++ b/__tests__/host.spec.ts
@@ -61,8 +61,9 @@ test("LanguageServiceHost", async () => {
 	expect(host.readFile(nonExistent)).toBeFalsy();
 	expect(host.readFile(testFile)).toEqual(unaryFunc);
 	expect(host.useCaseSensitiveFileNames()).toBe(process.platform === "linux");
-	expect(host.realpath(testFile)).toEqual(testFile);
-	expect(host.realpath(linkedTestFile)).toEqual(testFile);
+	// test realpath w/ symlinks. this returns a host path, so expect path.normalize()
+	expect(host.realpath(testFile)).toEqual(path.normalize(testFile));
+	expect(host.realpath(linkedTestFile)).toEqual(path.normalize(testFile));
 
 	// test misc functionality
 	expect(host.getCompilationSettings()).toEqual(testOpts);

--- a/src/host.ts
+++ b/src/host.ts
@@ -104,6 +104,11 @@ export class LanguageServiceHost implements tsTypes.LanguageServiceHost
 		return tsModule.sys.fileExists(path);
 	}
 
+	public realpath(path: string): string
+	{
+		return tsModule.sys.realpath!(path); // this exists in the default implementation: https://github.com/microsoft/TypeScript/blob/ab2523bbe0352d4486f67b73473d2143ad64d03d/src/compiler/sys.ts#L1288
+	}
+
 	public getTypeRootsVersion(): number
 	{
 		return 0;

